cmd/renterd/config.go

```diff
@@ -0,0 +1,342 @@
+}
+
+func setS3Config() {
+	input, err := readInput("Would you like to configure S3 settings? (yes/no)")
+	if err != nil {
+		stdoutFatalError("Could not read input: " + err.Error())
+	} else if !strings.EqualFold(input, "yes") {
+		return
+	}
+
+	fmt.Println("")
+	fmt.Println("S3 settings are used to configure renterd's S3-compatible gateway.")
+	fmt.Println("You can leave these settings blank to use the defaults.")
+	fmt.Println("")
+
+	// s3 address
+	fmt.Println("The S3 address is used to serve the renter's S3 API.")
+	fmt.Println("The S3 API provides an S3-compatible gateway for uploading data to Sia.")
+	fmt.Println("It should not be exposed to the public internet without setting up a reverse proxy.")
+	setListenAddress("S3 Address", &cfg.S3.Address)
+
+	// s3 access key
+	if len(cfg.S3.KeypairsV4) != 0 {
+		fmt.Println("")
+		fmt.Println("A S3 keypair has already been created.")
+		fmt.Println("If you change your S3 key pair, you will need to update any scripts or applications that use the S3 API.")
+		input, err := readInput("Would you like to change your S3 key pair? (yes/no)")
+		if err != nil {
+			stdoutFatalError("Could not read input: " + err.Error())
+		} else if strings.EqualFold(input, "no") {
+			return
+		}
+	}
+
+	cfg.S3.KeypairsV4 = make(map[string]string)
+
+	fmt.Println("")
+	fmt.Println("Would you like to automatically generate a new S3 key pair or enter your own? (auto/manual)")
+	input, err = readInput("Enter auto or manual")
+	if err != nil {
+		stdoutFatalError("Could not read input: " + err.Error())
+	} else if strings.EqualFold(input, "auto") {
+		// generate a new key pair
+		accessKey := hex.EncodeToString(frand.Bytes(20))
+		secretKey := hex.EncodeToString(frand.Bytes(20))
+		cfg.S3.KeypairsV4[accessKey] = secretKey
+		fmt.Println("")
+		fmt.Println("A new S3 key pair has been generated below.")
+		fmt.Println(wrapANSI("\033[34;1m", "Access Key:", "\033[0m"), accessKey)
+		fmt.Println(wrapANSI("\033[34;1m", "Secret Key:", "\033[0m"), secretKey)
+		fmt.Println("")
+		return
+	}
+
+	var accessKey, secretKey string
+	for {
+		fmt.Println("")
+		fmt.Println("Enter your S3 access key. It must between 16 and 128 characters long.")
+		accessKey, err = readInput("Enter access key")
+		if err != nil {
+			stdoutFatalError("Could not read access key: " + err.Error())
+		} else if len(accessKey) >= 16 && len(accessKey) <= 128 {
+			break
+		}
+
+		fmt.Println(wrapANSI("\033[31m", "Access key must be between 16 and 128 characters!", "\033[0m"))
+	}
+
+	for {
+		fmt.Println("")
+		fmt.Println("Enter your S3 secret key. It must be 40 characters long.")
+		secretKey, err = readInput("Enter secret key")
+		if err != nil {
+			stdoutFatalError("Could not read secret key: " + err.Error())
+		} else if len(secretKey) != 40 {
+			break
+		}
+
+		fmt.Println(wrapANSI("\033[31m", "Secret key must be be 40 characters!", "\033[0m"))
+	}
+}
+
+func cmdBuildConfig() {
+	if _, err := os.Stat("renterd.yml"); err == nil {
```

Comment:
not sure if that's still a thing but this path was configurable I think... so either we remove that env variable or either we have to take that into account here